lib/galaxy/datatypes/binary.py

```diff
@@ -614,7 +640,17 @@ def get_chunk(self, trans, dataset: "DatasetInstance", offset: int = 0, ck_size:
             offset = -1
         return dumps({"ck_data": util.unicodify(ck_data), "offset": offset})
 
-    def display_data(self, trans, dataset, preview=False, filename=None, to_ext=None, offset=None, ck_size=None, **kwd):
+    def display_data(
+        self,
+        trans,
+        dataset: "HistoryDatasetAssociation",
```

Comment:
Somewhere in the definition of `display_data` in the parent class, there's a reference to `dataset.hid` - so that makes it an HDA (no hid on a DatasetInstance).

Reply:
Remove that then ? It makes no sense, and you should be able to use this from e.g. the library.